channels/smartcard/client/smartcard_pack.c

```diff
@@ -850,7 +850,7 @@ void smartcard_trace_list_readers_return(SMARTCARD_DEVICE* smartcard, ListReader
 		CopyMemory(mszA, ret->msz, ret->cBytes);
 	}
 
-	for (index = 0; index < length - 2; index++)
+	for (index = 0; index < length - 1; index++)
```

Comment:
index should be ```size_t``` just like length.
There are some remaining casts of ```lenght``` to ```int``` which are no longer required (can't comment as not really in changed area)